modulemd/v2/tests/ModulemdTests/defaults.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/python3
+
+# This file is part of libmodulemd
+# Copyright (C) 2018 Red Hat, Inc.
+#
+# Fedora-License-Identifier: MIT
+# SPDX-2.0-License-Identifier: MIT
+# SPDX-3.0-License-Identifier: MIT
+#
+# This program is free software.
+# For more information on the license, see COPYING.
+# For more information on free software, see
+# <https://www.gnu.org/philosophy/free-sw.en.html>.
+
+import sys
+try:
+    import unittest
+    import gi
+    gi.require_version('Modulemd', '2.0')
+    from gi.repository import Modulemd
+except ImportError:
+    # Return error 77 to skip this test on platforms without the necessary
+    # python modules
+    sys.exit(77)
+
+from base import TestBase
+
+
+class TestDefaults(TestBase):
+
+    def test_constructors(self):
+        # Test that the new() function works
+        defs = Modulemd.Defaults.new(
+            Modulemd.DefaultsVersionEnum.ONE, 'foo')
+        assert defs
+
+        assert defs.props.mdversion == Modulemd.DefaultsVersionEnum.ONE
+        assert defs.get_mdversion() == Modulemd.DefaultsVersionEnum.ONE
+
+        assert defs.props.module_name == 'foo'
+        assert defs.get_module_name() == 'foo'
+
+        # Test that we cannot instantiate directly
+        try:
+            Modulemd.Defaults()
+            assert False
+        except TypeError as e:
+            assert 'cannot create instance of abstract' in e.__str__()
+
+        # Test with a zero mdversion
+        try:
+            with self.expect_signal():
+                defs = Modulemd.Defaults.new(0, 'foo')
+            assert False
+        except TypeError as e:
+            assert 'constructor returned NULL' in e.__str__()
+
+        # Test with an unknown mdversion
+        try:
+            with self.expect_signal():
+                defs = Modulemd.Defaults.new(
+                    Modulemd.DefaultsVersionEnum.LATEST + 1, 'foo')
+            assert False
```

Comment:
Perhaps instead of this `try:` block, use `self.assertRaises`?
For an example, see the `Profile` patch: https://github.com/fedora-modularity/libmodulemd/pull/113/files#diff-3a8af7909cf33247e7740e17b3bcfb58R60.

Reply:
Ah, I didn't see that in the other patch. Handy!

Reply:
Replaced all of these cases with self.assertRaisesRegex() so I can also verify that the failure message is correct.